src/core/Akka.Streams/ActorMaterializer.cs

```diff
@@ -25,6 +25,53 @@
 
 namespace Akka.Streams
 {
+    /// <summary>
+    /// INTERNAL API
+    /// </summary>
+    internal sealed class DefaultMaterializerExt : ExtensionIdProvider<DefaultMaterializer>
+    {
+        public override DefaultMaterializer CreateExtension(ExtendedActorSystem system)
+        {
+            return new DefaultMaterializer(system);
+        }
+    }
+    
+    /// <summary>
+    /// INTERNAL API
+    /// </summary>
+    /// <remarks>
+    /// Caches a default materializer instance for each actor system. Prevents the need to create a new materializer
+    /// for trivial changes.
+    /// </remarks>
+    internal sealed class DefaultMaterializer : IExtension
```

Comment:
I created an `ActorSystemExtension` here which does the following:

1. Ensures that the Akka.Streams HOCON only needs to be injected once per-ActorSystem - this allowed me to eliminate all of the caching code I had to add in https://github.com/akkadotnet/akka.net/pull/6440 since that's no longer needed.
2. Re-uses the same `ActorMaterializer` when we're materializing a top-level stream using all of the defaults.